### PR TITLE
TRUNK-5119: Update apache velocity

### PR DESCRIPTION
**Pull request by gitcliff**
link:https://issues.openmrs.org/browse/TRUNK-5119

TRUNK-5119: Update apache velocity

### CODE REVIEW

File: api/pom.xml [hunk 1]
```diff
@@ -35,6 +35,10 @@
          <groupId>org.apache.commons</groupId>
          <artifactId>commons-lang3</artifactId>
       </dependency>
+      <dependency>
```
Comment (dkayiwa) [inline api/pom.xml]: Is this change related to updating apache velocity?

File: api/src/main/java/org/openmrs/util/VelocityExceptionHandler.java [hunk 2]
```diff
@@ -43,4 +44,10 @@ public Object methodException(Class claz, String method, Exception e) throws Exc
 		throw e;
 	}
 	
+	@Override
+	public Object methodException(Context context, Class claz, String method, Exception e, Info info) {
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/util/VelocityExceptionHandler.java]: Is this change related to updating apache velocity?

File: web/src/main/java/org/openmrs/web/filter/StartupFilter.java [hunk 3]
```diff
@@ -173,11 +173,10 @@ else if (!httpRequest.getServletPath().equals("/" + WebConstants.SETUP_PAGE_URL)
 	private void initializeVelocity() {
 		if (velocityEngine == null) {
 			velocityEngine = new VelocityEngine();
-			
 			Properties props = new Properties();
-			props.setProperty(RuntimeConstants.RUNTIME_LOG, "startup_wizard_vel.log");
+			props.setProperty(RuntimeConstants.RUNTIME_LOG_NAME, "startup_wizard_vel.log");
```
Comment (dkayiwa) [inline web/src/main/java/org/openmrs/web/filter/StartupFilter.java]: Is this change related to updating apache velocity?

File: web/src/main/java/org/openmrs/web/filter/StartupFilter.java [hunk 4]
```diff
@@ -196,6 +195,11 @@ private void initializeVelocity() {
 		}
 	}
 	
+	private Runtime Runtime() {
+		// TODO Auto-generated method stub
```
Comment (dkayiwa) [inline web/src/main/java/org/openmrs/web/filter/StartupFilter.java]: Is this change related to updating apache velocity?

File: web/src/main/java/org/openmrs/web/filter/StartupFilter.java (hunk 3 above)
Comment (gitcliff) [reply web/src/main/java/org/openmrs/web/filter/StartupFilter.java]: yes @dkayiwa  the new version of apache velocity 2.0 doesn't have RUNTIME_LOG in its API as compared to the older version of 1.6.2

File: web/src/main/java/org/openmrs/web/filter/StartupFilter.java (hunk 4 above)
Comment (gitcliff) [reply web/src/main/java/org/openmrs/web/filter/StartupFilter.java]: @dkayiwa ,,it was as a result of fixing a compilation error that occurred during the maven build after updating apache velocity

File: api/pom.xml (hunk 1 above)
Comment (gitcliff) [reply api/pom.xml]: @dkayiwa yes apache velocity 2.0 necessitated also dependency on commons-lang3

File: api/src/main/java/org/openmrs/util/VelocityExceptionHandler.java (hunk 2 above)
Comment (gitcliff) [reply api/src/main/java/org/openmrs/util/VelocityExceptionHandler.java]: yes @dkayiwa  because the updating the apache velocity made changes in the code and brought about some errors during compilation that needed to be fixed

File: api/src/main/java/org/openmrs/util/VelocityExceptionHandler.java [hunk 5]
```diff
@@ -43,4 +44,10 @@ public Object methodException(Class claz, String method, Exception e) throws Exc
...
+		// TODO Auto-generated method stub
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/util/VelocityExceptionHandler.java]: What is the use of the above comment?

Comment (gitcliff) [reply api/src/main/java/org/openmrs/util/VelocityExceptionHandler.java]: sorry @dkayiwa  for responding this LATE, have just seen it now.
the above comment "TODO Auto-generated method stub" was just an auto generated comment

File: web/src/main/java/org/openmrs/web/filter/StartupFilter.java (hunk 4 above)
Comment (gitcliff) [reply web/src/main/java/org/openmrs/web/filter/StartupFilter.java]: hello @dkayiwa the above change was due to the updating of apache velocity from 1.6 to 2.0 velocity core engine

File: api/src/main/java/org/openmrs/util/VelocityExceptionHandler.java (hunk 5 above)
Comment (Ruhanga) [reply api/src/main/java/org/openmrs/util/VelocityExceptionHandler.java]: @gitcliff,  auto generated from where? In any case, it is not advisable to submit code containing any TODO. So why have a duplicate method `methodException(Context context, Class claz, String method, Exception e, Info info)` for the changes made in this class?

Comment (Ruhanga) [reply api/src/main/java/org/openmrs/util/VelocityExceptionHandler.java]: Also checkout this https://wiki.openmrs.org/display/docs/Java+Conventions .

**Comment by mozzy11** [comment]
@gitcliff  <version>2.1</version> is the latest Full release Version. but You seem to have not responded to the comments. can you revise the comments again?

File: api/src/main/java/org/openmrs/util/VelocityExceptionHandler.java (hunk 5 above)
Comment (gitcliff) [reply api/src/main/java/org/openmrs/util/VelocityExceptionHandler.java]: okay @Ruhanga thanks for this,,,going to check it out

**Comment by mozzy11** [comment]
@gitcliff have you seen the merge conflicts??
also Properly format your your commit message

File: web/src/main/java/org/openmrs/web/filter/StartupFilter.java [hunk 6]
```diff
@@ -173,11 +173,10 @@ else if (!httpRequest.getServletPath().equals("/" + WebConstants.SETUP_PAGE_URL)
 	private void initializeVelocity() {
 		if (velocityEngine == null) {
 			velocityEngine = new VelocityEngine();
-			
 			Properties props = new Properties();
-			props.setProperty(RuntimeConstants.RUNTIME_LOG, "startup_wizard_vel.log");
+			props.setProperty(RuntimeConstants.RUNTIME_LOG_NAME, "startup_wizard_vel.log");
 			// Linux requires setting logging properties to initialize Velocity Context.
-			props.setProperty(RuntimeConstants.RUNTIME_LOG_LOGSYSTEM_CLASS,
+			props.setProperty(RuntimeConstants.RUNTIME_LOG_INSTANCE,
```
Comment (mozzy11) [inline web/src/main/java/org/openmrs/web/filter/StartupFilter.java]: what was behind changing the name here ??

File: web/src/main/java/org/openmrs/web/filter/StartupFilter.java (hunk 3 above)
Comment (mozzy11) [inline web/src/main/java/org/openmrs/web/filter/StartupFilter.java]: why change this?

**Comment by mozzy11** [comment]
@gitcliff  , have you seen the merge conflicts??
Did you see my comment  regarding Merge conficts  here https://talk.openmrs.org/t/upgrade-core-libraries-sprint-announcement/23193/17 ??
